galasa-ui/src/components/test-runs/test-run-details/3270Tab/TableOfScreenshots.tsx

```diff
@@ -0,0 +1,215 @@
+    ) as TerminalImage;
+    console.log('Table row clicked! Image data:\n' + JSON.stringify(newImageData));
+    setImageData(newImageData);
+  };
+
+  const terminalnames = useMemo(() => {
+    const names = new Set(flattenedZos3270TerminalData.map((row: CellFor3270) => row.Terminal));
+    return [
+      { id: 'all', label: 'All' },
+      ...Array.from(names).map((name) => ({ id: name, label: name })),
+    ];
+  }, [flattenedZos3270TerminalData]);
+
+  const filteredRows = useMemo(() => {
+    let result: CellFor3270[] = flattenedZos3270TerminalData;
+
+    if (searchTerm) {
+      const lowerCaseSearchTerm = searchTerm.toLowerCase();
+      result = result.filter((row: CellFor3270) =>
+        Object.values(row).some((value) =>
+          String(value).toLowerCase().includes(lowerCaseSearchTerm)
+        )
+      );
+    }
+
+    if (selectedTerminal && selectedTerminal.id !== 'all') {
+      result = result.filter((row) => row.Terminal === selectedTerminal.id);
+    }
+    // console.log("Hello\n" + JSON.stringify(result));
+
+    return result;
+  }, [searchTerm, selectedTerminal, flattenedZos3270TerminalData]);
+
+  useEffect(() => {
+    // Ensure screenshots are only collected once.
+    if (!screenshotsCollected) {
+      screenshotsCollected = true;
+
+      setIsLoading(true);
+      const fetchData = async () => {
+        try {
+          setFlattenedZos3270TerminalData([]);
+          setAllImageData([]);
+          const { newFlattenedZos3270TerminalData, newAllImageData } = await get3270Screenshots(
+            zos3270TerminalData,
+            runId
+          );
+          setFlattenedZos3270TerminalData(newFlattenedZos3270TerminalData);
+          setAllImageData(newAllImageData);
+          setIsLoading(false);
+        } catch (error) {
+          console.error('Error fetching data:', error);
+          setIsError(true);
+          setIsLoading(false);
+        }
+      };
+
+      fetchData();
+    }
+  }, []);
+
+  if (isLoading) {
+    return (
+      <DataTableSkeleton
+        data-testid="loading-table-skeleton"
+        columnCount={headers.length}
+        rowCount={15}
+      />
+    );
+  }
+
+  return (
+    <DataTable isSortable rows={filteredRows} headers={headers} stickyHeader>
+      {({
+        rows,
+        headers,
+        getTableProps,
+        getHeaderProps,
+        getRowProps,
+      }: {
+        rows: DataTableRow[];
+        headers: DataTableHeader[];
+        getHeaderProps: (options: any) => TableHeadProps;
+        getRowProps: (options: any) => TableRowProps;
+        getTableProps: () => TableBodyProps;
+      }) => (
+        <TableContainer>
+          <TableToolbarContent>
+            <TableToolbarSearch
+              placeholder={translations('searchPlaceholder')}
+              persistent
+              onChange={(e: any) => {
+                setSearchTerm(e.target.value);
+              }}
+              data-testid="search-input"
+            />
+            <Dropdown
```

Comment:
Running this locally, I noticed that the dropdown appears even if there's only one terminal in a test. The options are "All" and "term1" - is that the expected behaviour? I feel like it would make more sense to only show the dropdown if there are multiple terminals to filter by.

Also, the dropdown's content seems to appear behind the actual table:

<img width="693" height="526" alt="Screenshot 2025-08-28 at 13 12 57" src="https://github.com/user-attachments/assets/3e58dfe7-2929-47c4-82f1-ee4817bde8f1" />